src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -970,11 +975,32 @@ public String getName() {
         @Override
         public GroupDetailsImpl mapAttributes(String dn, Attributes attributes) throws NamingException {
             LdapName name = new LdapName(dn);
-            String groupName = fixGroupname(String.valueOf(name.getRdn(name.size() - 1).getValue()));
+            String groupName = fixGroupname(extractGroupName(name, attributes));
             return new GroupDetailsImpl(dn, groupName);
         }
-    }
 
+        static String extractGroupName(LdapName name, Attributes attributes) throws NamingException {
+            final String CN = "cn";
+            String groupName = StringUtils.EMPTY;
+            if (!GET_CN_FROM_ATTRIBUTES) {
+                groupName = String.valueOf(name.getRdn(name.size() - 1).getValue());
+            } else {
+                for (NamingEnumeration ae = attributes.getAll(); ae.hasMore(); ) {
+                    Attribute attr = (Attribute) ae.next();
+                    if (CN.equals(attr.getID())) {
+                        for (NamingEnumeration e = attr.getAll(); groupName.equals(StringUtils.EMPTY);) {
+                            groupName = e.next().toString();
+                            if (e.hasMore()) {
+                                LOGGER.log(Level.WARNING,"The group " + name.getRdns() + " has more than one cn value. The first one  (" + groupName + ") has been assigned as external group name");
```

Comment:
I would hesitate to log this at WARNING continuously. If the schema has a group like this, then every time the group is loaded this message will be printed, which seems pretty spammy. Maybe only log once, or move it to `Level.FINE` so that it can be used as a diagnostic tool if things aren't working but isn't printed otherwise.

Reply:
I have moved it to `Level.FINE`